controllers/userC.js

```diff
@@ -22,5 +22,27 @@ const createUser = asyncWrapper(async(req, res, next) => {
 
     res.status(201).json({ msg: 'User created successfully', user: newUser });
 });
+ const userLogin = asyncWrapper(async (req, res, next) => {
+    const userLoginRequest = { email: req.body.email, password: req.body.password };
+    const user = await fetchUser({ email: req.body.email });
+    if (!user) {
+        return next(createCustomError("User not found", 404));
+    }
+
+    const correctPassword = await bcrypt.compare(userLoginRequest.password, user.password );
+    if (correctPassword === false) {
+        return next(createCustomError('Invalid email or password', 404));
+    }
+    const generatedToken = jwt.sign({
+        id: user._id,
+        email: user.email
+    }, process.env.TOKEN, { expiresIn: '12h' });
+    const result = {
+        id: user._id,
+        email: user.email,
+        token: generatedToken
+    }
+    return res.status(200).json({ msg: "userLogin successful", result });
+ });
 
-module.exports = { createUser }
\ No newline at end of file
+module.exports = { createUser, userLogin }
```

Comment:
Add an extra line.
Also ensure you test this implementation

Reply:
Okay, noted